tensorflow_probability/python/distributions/gev.py

```diff
@@ -0,0 +1,265 @@
+from tensorflow_probability.python.distributions import uniform
+from tensorflow_probability.python.internal import distribution_util
+from tensorflow_probability.python.internal import dtype_util
+from tensorflow_probability.python.internal import tensor_util
+
+
+class GeneralizedExtremeValue(transformed_distribution.TransformedDistribution):
+  """The scalar GeneralizedExtremeValue distribution
+  with location `loc`, `scale` and `concentration` parameters.
+
+  #### Mathematical details
+
+  The probability density function (pdf) of this distribution is,
+
+  ```none
+  pdf(x; loc, scale, conc) = t(x; loc, scale, conc) ** (1 + conc) * exp(
+  - t(x; loc, scale, conc) ) / scale
+  where t(x) =
+    * (1 + conc * (x - loc) / scale) ) ** (-1 / conc) when conc != 0;
+    * exp(-(x - loc) / scale) when conc = 0.
+  ```
+
+  where `concentration = conc`.
+
+  The cumulative density function of this distribution is,
+
+  ```cdf(x; mu, sigma) = exp(-t(x))```
+
+  The generalized extreme value distribution is a member of the
+  [location-scale family](https://en.wikipedia.org/wiki/Location-scale_family),
+  i.e., it can be constructed as,
+
+  ```none
+  X ~ GeneralizedExtremeValue(loc=0, scale=1, concentration=conc)
+  Y = loc + scale * X
+  ```
+
+  #### Examples
+
+  Examples of initialization of one or a batch of distributions.
+
+  ```python
+  tfd = tfp.distributions
+
+  # Define a single scalar generalized extreme values distribution.
+  dist = tfd.GeneralizedExtremeValue(loc=0., scale=3., concentration=0.9)
+
+  # Evaluate the cdf at 1, returning a scalar.
+  dist.cdf(1.)
+
+  # Define a batch of two scalar valued generalized extreme values.
+  # The first has mean 1 and scale 11, the second 2 and 22.
+  dist = tfd.GeneralizedExtremeValue(loc=[1, 2.], scale=[11, 22.])
+
+  # Evaluate the pdf of the first distribution on 0, and the second on 1.5,
+  # returning a length two tensor.
+  dist.prob([0, 1.5])
+
+  # Get 3 samples, returning a 3 x 2 tensor.
+  dist.sample([3])
+  ```
+
+  Arguments are broadcast when possible.
+
+  ```python
+  # Define a batch of two scalar valued Logistics.
+  # Both have mean 1, but different scales.
+  dist = tfd.GeneralizedExtremeValue(loc=1., scale=1, concentration=[0, 0.9])
+
+  # Evaluate the pdf of both distributions on the same point, 3.0,
+  # returning a length 2 tensor.
+  dist.prob(3.0)
+  ```
+
+  """
+
+  def __init__(self,
+               loc,
+               scale,
+               concentration,
+               validate_args=False,
+               allow_nan_stats=True,
+               name='GeneralizedExtremeValue'):
+    """Construct generalized extreme value distributions with location `loc`,
+    scale `scale`, and concentration `concentration`.
+
+    The parameters `loc`, `scale`, and `concentration` must be shaped in a way
+    that supports broadcasting (e.g. `loc + scale` + `concentration` is valid).
+
+    Args:
+      loc: Floating point tensor, the means of the distribution(s).
```

Comment:
It's not the actual mean. Maybe just call it "the location parameter of the distribution(s)"

Reply:
Thx for pointing this out, I will also fix all other places of misusing mean and loc.